clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -777,6 +782,23 @@ constructKernelName(Sema &S, FunctionDecl *KernelCallerFunc,
                                       KernelNameType)};
 }
 
+static bool hasNativeSycl2020SpecConstantSupport(ASTContext &Context) {
```

Comment:
I'd probably prefer something more generic here for the name, like "IsNativeSomethingOrOther".

Reply:
I do not know what the generic name for targets which satisfy `T.isSPIR() && T.getSubArch() == llvm::Triple::NoSubArch` is. @bader could you please suggest something

Reply:
I spoke to @mdtoguchi offline and he suggested `isDefaultSYCLArch()` or `isDefaultSPIRArch()`. I plan to go with `isDefaultSPIRArch()` unless someone here objects :)

Reply:
What is the meaning of `isDefaultSPIRArch()` for implementations not using SPIR, like CUDA or others, where they could be an equivalent of specialization constant?

Reply:
It was my understanding that non-SPIR targets do not have specialization constants support and require special processing. `isDefaultSPIRArch() `  is used to determine when this special processing is required for specialization constants. I do not what the equivalents in CUDA etc are and how/if they are handled in SYCL compiler. @AlexeySachkov @romanovvlad @bader could you please answer?